### PR TITLE
ssh: split out conf files, install sysvinit startup files in openssh-sshd pkg

### PR DESCRIPTION
**Pull request by sknsean**

### CODE REVIEW

File: recipes/openssh/openssh.inc [hunk 1]
```diff
@@ -70,15 +70,18 @@ do_install_extra_files() {
...
+FILES_${PN}-ssh = "${bindir}/slogin"
 FILES_${PN}-ssh-doc = "${mandir}/*/ssh_config.* ${mandir}/*/slogin.*"
 FILES_${PN}-sshd = "${sysconfdir}/init.d/sshd \
-	${sysconfdir}/ssh/moduli ${sysconfdir}/ssh/sshd_config \
+	${sysconfdir}/rcS.d/* \
+	${sysconfdir}/ssh/moduli \
 	${localstatedir}/run/sshd"
 FILES_${PN}-sshd-doc = "${mandir}/*/moduli.* ${mandir}/*/sshd_config.*"
+FILES_${PN}-ssh-config = "${sysconfdir}/ssh/ssh_config"
+FILES_${PN}-sshd-config = "${sysconfdir}/ssh/sshd_config"
```
Comment (kimrhh) [inline recipes/openssh/openssh.inc]: you should make the ssh package depend on these two new packages to make sure we don't break stuff downstream. Probably best to let the new packages PROVIDE sshd_config and ssh_config respectively, and then depend on that, so it becomes easy to override those with your own little recipe.

File: recipes/openssh/openssh.inc [hunk 2]
```diff
@@ -72,13 +72,20 @@ inherit auto-package-utils
 AUTO_PACKAGE_UTILS = "ssh-add ssh-agent ssh-keyscan ssh-keygen ssh-keysign \
 	scp ssh sshd sftp sftp-server ssh-pkcs11-helper"
 
+PACKAGES_${PN} += "ssh-config sshd-config"
```
Comment (esben) [inline recipes/openssh/openssh.inc]: Although not really a requirement, normally we prefix all packages with ${PN}, and I don't see a special reason to do it different here.

Comment (sknsean) [inline recipes/openssh/openssh.inc]: fixed

**Comment by esben** [comment]
LGTM :+1:

File: recipes/openssh/openssh.inc [hunk 3]
```diff
@@ -72,13 +72,20 @@ inherit auto-package-utils
...
 FILES_${PN}-sshd = "${sysconfdir}/init.d/sshd \
-	${sysconfdir}/ssh/moduli ${sysconfdir}/ssh/sshd_config \
+	${sysconfdir}/rcS.d/* \
+	${sysconfdir}/ssh/moduli \
 	${localstatedir}/run/sshd"
 FILES_${PN}-sshd-doc = "${mandir}/*/moduli.* ${mandir}/*/sshd_config.*"
+FILES_${PN}-ssh-config = "${sysconfdir}/ssh/ssh_config"
+PROVIDES_${PN}-ssh-config = "ssh-config"
+FILES_${PN}-sshd-config = "${sysconfdir}/ssh/sshd_config"
+PROVIDES_${PN}-sshd-config = "sshd-config"
```
Comment (sknsean) [inline recipes/openssh/openssh.inc]: Is this PROVIDES needed?

Comment (esben) [inline recipes/openssh/openssh.inc]: No, they are actually both not really needed. You can just provide openssh-ssh-config and/or openssh-sshd-config, which is probably better, as such packages are very unlikely to work with other ssh packages than openssh.

**Comment by kimrhh** [comment]
merged, thanks

**Pull request merged**